datalad/downloaders/base.py

```diff
@@ -176,6 +176,12 @@ def access(self, method, url, allow_old_session=True, **kwargs):
                 break
             except AccessDeniedError as e:
                 ce = CapturedException(e)
+                if hasattr(e, 'status') and e.status == 429:
```

Comment:
and for `429` we get AccessDeniedError and not DownloadError which is handled below?  
do we have logging when it happens or we might want to add an explicit `lgr.debug("` while treating it with sleep

Reply:
> and for 429 we get AccessDeniedError and not DownloadError which is handled below?

Yes. We raise `AccessDeniedError` for any code in 400..500 range. And to some extend the notion of "denied" is alright. It's just that this part here regards that as authentication related which it is not.

Reply:
> do we have logging when it happens 

`CapturedException` logs the exception and there's level 5 log on every attempt. I'd prefer to not add more. In my view it's already hard enough to find the relevant messages in the logs. If everything is repeated and rephrased at higher log levels, things get just harder to read, I think.

Reply:
ok